src/pages/SignTransaction/components/SpendLimitInfo/TokenSpendLimit.tsx

```diff
@@ -133,13 +133,7 @@ export function TokenSpendLimit({
             displayValue: isInfinite
               ? t('Unlimited')
               : (diffItemValue as TokenUnit).toDisplay(),
-            usdPrice:
-              isInfinite || !approval.usdPrice
-                ? undefined
-                : String(
-                    (diffItemValue as TokenUnit).toDisplay({ asNumber: true }) *
-                      Number(approval.usdPrice)
-                  ),
+            usdPrice: isInfinite ? undefined : approval.usdPrice,
```

Comment:
This was being multiplied unnecessarily - I took `usdPrice` as the price of a single token here, but it's the USD value of the whole approval.